### ISSUE TITLE
Problem: Tendermint got stuck without any log afterward in linux/amd64 Ubuntu 18.04.4 LTS

### ISSUE THREAD

**Issue by allthatjazzleo** [tendermint/tendermint#5162]
### What version of Tendermint are you using?
$ ./tendermint --version
0.33.6-606d0a89
### Does this issue reproduce with the latest release?
Yes.

### What operating system and processor architecture are you using?
Ubuntu 18.04.4 LTS linux/amd64

### What did you do?
Download tendermint binary from release page and run with our abci application

### What did you expect to see?
tendermint is nominal.

### What did you see instead?
tendermint get stuck after a few hours or days. They were using 100% CPU of a single core and had log like

<details><summary>Tendermint Log</summary>

```bash
tendermint[22671]: I[2020-07-09|15:33:22.714] Committed state                              module=state height=217547 txs=0 appHash=1042D0B36865D9514993ABA1641093B39A01BF33F085FFE8D404E0BB46ACEE21
tendermint[22671]: I[2020-07-09|15:34:14.008] Executed block                               module=state height=217548 validTxs=1 invalidTxs=0
tendermint[22671]: I[2020-07-09|15:34:14.019] Committed state                              module=state height=217548 txs=1 appHash=D60792F4D2A9AE26E176CCEBE9B61B2280D5C2BDA3A6395614D6CC16150B66B4
```

No log generated after stuck happen
</details>

The rpc server is also not responding.

What's more, the stuck only happen in some vm in cloud provider we tested:
* Standard DC2s_v2 (Azure)
* t2.medium (AWS)
  
However, it didn't happen in *Bare metal server Intel Xeon E3-1270 v6* (IBM) and *Standard_DC2s* (Azure) 

We found the issue possibly related is https://github.com/golang/go/issues/38223

Not sure if tendermint is built by go1.14.1 in release page which could create stuck issue.

### What did you solve instead?
We have to clone the repo and build the tendermint in those affected VM using go1.13.3.

It runs without stuck issue.

**Comment by marbar3778** [comment]
thanks for opening this issue. 0.33.6 was built with 1.14.4 when I checked circleci/golang release schedule. If you build the repo with go1.14.4 does the error persist?

**Comment by allthatjazzleo** [comment]
@marbar3778 
But I docker pull the image `tendermintdev/docker-tendermint-build` which is used to build the binary in circleci
and check the go version
```bash
$ docker run -it tendermintdev/docker-tendermint-build bash
circleci@a4746f46943c:/go$ go version
go version go1.14.1 linux/amd64
```
it returns `go1.14.1 linux/amd64`

**Comment by marbar3778** [comment]
I went to the circleci documentation and read 1.14.4, sorry for the misinformation. I'm not sure if we are able to change the binaries on the release page.

**Comment by erikgrinaker** [comment]
This seems like a fairly important issue, we can consider making a new release with updated binaries.

**Comment by marbar3778** [comment]
I am attempting to rebuild the docker image with a newer version of go, once this is completed we can work on a new release. It may be beneficial to include the privval fix as well

**Comment by erikgrinaker** [comment]
We've released 0.33.7 which should fix this. Thanks for reporting, let us know if you still experience any issues.
